src/sources/repeat_with.rs

```diff
@@ -44,7 +44,7 @@ where
     #[inline]
     fn next(&mut self) -> Option<<Self as Lending<'_>>::Lend> {
         // SAFETY: 'a: 'lend
-        Some(unsafe { core::mem::transmute::<<Self as Lending<'a>>::Lend, <Self as Lending<'_>>::Lend>((self.f)()) })
+        Some(unsafe { core::mem::transmute::<<L as Lending<'a>>::Lend, <L as Lending<'_>>::Lend>((self.f)()) })
```

Comment:
> I'm really confused. The commit seem to have nothing to do with the post. Am I missing something?

@vigna Before this change, the full annotated type here was `<Self as Lending<'a, &'a Self>>::Lend`, which requires `&'a Self` to be a "well-formed" reference type, which requires `Self: 'a` => `RepeatWith<'a, L, F>: 'a` => (`F: 'a` and `L: 'a`).
`L: 'a` is satified by the where clause, but `F: 'a` is not hence the error.

After this change, the full type is `<L as Lending<'a, &'a L>>::Lend`. This only requires `L: 'a` which is already satisfied by the bounds.